core/src/test/scala/org/apache/spark/util/UninterruptibleThreadSuite.scala

```diff
@@ -80,130 +97,128 @@ class UninterruptibleThreadSuite extends SparkFunSuite {
-              hasInterruptedException = true
+          hasInterruptedException = await(interruptLatch)
+          if (!hasInterruptedException) {
+            runUninterruptibly {
+              hasInterruptedException = sleep(0)
             }
-          }
-          if (sleep(1)) {
-            hasInterruptedException = true
+            hasInterruptedException |= sleep(0)
           }
         }
         interruptStatusBeforeExit = Thread.interrupted()
       }
     }
     t.start()
-    assert(enterRunUninterruptibly.await(10, TimeUnit.SECONDS), "await timeout")
+    assert(!await(enterRunUninterruptibly), "await interrupted")
     t.interrupt()
     interruptLatch.countDown()
+    assert(!sleep(0), "sleep should not be interrupted")
+    t.interrupt()
+    assert(!sleep(0), "sleep should not be interrupted")
+    t.interrupt()
     t.join()
-    assert(hasInterruptedException === false)
-    assert(interruptStatusBeforeExit)
+    assert(!hasInterruptedException, "runUninterruptibly should not be interrupted")
+    assert(interruptStatusBeforeExit, "interrupt flag should be set")
   }
 
   test("no runUninterruptibly") {
     @volatile var hasInterruptedException = false
-    val t = new UninterruptibleThread("test") {
+    @volatile var interruptStatusBeforeExit = false
+    val t = new UninterruptibleThread("run") {
       override def run(): Unit = {
-        if (sleep(0)) {
-          hasInterruptedException = true
-        }
+        hasInterruptedException = sleep(0)
+        interruptStatusBeforeExit = Thread.interrupted()
       }
     }
     t.interrupt()
     t.start()
     t.join()
-    assert(hasInterruptedException === true)
+    assert(hasInterruptedException, "run should be interrupted")
+    assert(!interruptStatusBeforeExit, "interrupt flag should not be set")
   }
 
   test("SPARK-51821 uninterruptibleLock deadlock") {
-    val latch = new CountDownLatch(1)
-    val task = new UninterruptibleThread("task thread") {
+    val interruptLatch = new CountDownLatch(1)
+    val t = new UninterruptibleThread("run") {
       override def run(): Unit = {
         val channel = new AbstractInterruptibleChannel() {
           override def implCloseChannel(): Unit = {
             begin()
-            latch.countDown()
+            interruptLatch.countDown()
             try {
               Thread.sleep(Long.MaxValue)
             } catch {
-              case _: InterruptedException => Thread.currentThread().interrupt()
+              case e: InterruptedException =>
+                log.info("sleep interrupted", e)
+                Thread.currentThread().interrupt()
             }
           }
         }
         channel.close()
       }
     }
-    task.start()
-    assert(latch.await(10, TimeUnit.SECONDS), "await timeout")
-    task.interrupt()
-    task.join()
+    t.start()
+    assert(!await(interruptLatch), "await interrupted")
+    t.interrupt()
+    t.join()
   }
 
   test("stress test") {
+    for (i <- 0 until 20) {
```

Comment:
Is this the major change tries to improve the test coverage? Since it's executed in sequence, the stress of interruptions to a single `UninterruptibleThread` doesn't seem to be inscreased to me.

Reply:
This change does not target to increase concurrency of the stress test. It targets to reproduce [SPARK-53394](https://issues.apache.org/jira/browse/SPARK-53394)